theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -566,12 +567,19 @@ def _step(
         previous_err: torch.Tensor,
         converged_indices: torch.Tensor,
         force_update: bool,
+        truncated_grad_loop: bool,
         **kwargs,
     ) -> Tuple[torch.Tensor, bool]:
         tensor_dict, err = self._compute_retracted_tensors_and_error(
             delta, converged_indices, force_update
         )
-        reject_indices = self._complete_step(delta, err, previous_err, **kwargs)
+        if truncated_grad_loop:
```

Comment:
Can't remember, is the implicit grad part of the truncated_grad_loop flag?

Reply:
Yes, `truncated_grad_loop` is the final attached loop of "TRUNCATED" and "IMPLICIT". I was thinking that this is kind of a confusing name tbh.

Reply:
Yeah we could rename to `implicit_and_trunc_grad_block` or maybe `..._loop` is okay. We can also add a comment to clarify.